### PR TITLE
Fixing search messages

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
### 🎯 Goal

Fixing search for messages.

I also fixed unit tests in state module so I can add new unit tests for the changed code. The unit tests unit JUNIT5 were not running.

### 🛠 Implementation details

- Scroll updates were always being set to true, which is just wrong. Fixed in this PR.
- Offline results were being used for new messages, which should never be used for newer messages. 

### 🎨 UI Changes

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="https://user-images.githubusercontent.com/10619102/177880586-2cb5f950-df7d-49c9-97fd-ad62cc2a7c04.mov" controls="controls" muted="muted" />
</td>
<td>
<video src="https://user-images.githubusercontent.com/10619102/177880389-cb386526-8139-4e51-a0f8-2bf757aadc1a.mov" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>

### 🧪 Testing

Search messages. You can use "secret message" inside Skinny Secret

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- ~[ ] PR is linked to the GitHub issue it resolves~

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- ~[ ] Affected documentation updated (KDocs, docusaurus, tutorial)~

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

giphy (1)

### CODE REVIEW

File: stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelLogic.kt [hunk 1]
```diff
@@ -216,11 +216,12 @@ internal class ChannelLogic(
     }
 
     private suspend fun runChannelQuery(request: WatchChannelRequest): Result<Channel> {
-        val offlineChannel = runChannelQueryOffline(request)
+        val offlineChannel = if (!request.isFilteringNewerMessages()) runChannelQueryOffline(request) else null
```
Comment (kanat) [inline stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelLogic.kt:219]: Hey @leandroBorgesFerreira,

Can we move this check right into `runChannelQueryOffline` method? :)
I can we see we do exact same check in `onQueryChannelRequest`.

**Review by kanat — APPROVED** [review]
LGTM

File: stream-chat-android-state/.env [hunk 2]
```diff
@@ -0,0 +1,3 @@
+STREAM_LOG_LEVEL=ALL
```
Comment (adasiewiczr) [inline stream-chat-android-state/.env:1]: Why do we need to keep that?

**Review by TGazica — APPROVED** [review]
Looks good on my side, unblocks #3818 and #3849

**Pull request merged**